### PR TITLE
AC: telemetry

### PR DESCRIPTION
**Pull request by eaidova**
TO DO:

- [x] converters usage
- [x] custom evaluators usage

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
12:46:25  accuracy_checker/utils.py:860: [E1123(unexpected-keyword-arg), init_telemetry] Unexpected keyword argument 'version' in constructor call

File: tools/accuracy_checker/accuracy_checker/utils.py [hunk 1]
```diff
@@ -847,3 +849,44 @@ def convert_xctr_yctr_w_h_to_x1y1x2y2(x, y, width, height):
...
     return x1, y1, x2, y2
+
+
+def init_telemetry():
+    try:
+        import openvino_telemetry as tm # pylint:disable=C0415
+    except ImportError:
+        return None
+    try:
+        telemetry = tm.Telemetry('Accuracy Checker', version=__version__, tid='UA-194864834-1')
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/utils.py:860]:
```suggestion
        telemetry = tm.Telemetry('Accuracy Checker', app_version= __version__, tid='UA-194864834-1')
```

File: tools/accuracy_checker/accuracy_checker/evaluators/model_evaluator.py [hunk 2]
```diff
@@ -196,6 +199,37 @@ def get_processing_info(config):
             dataset_config['name']
         )
 
+    def send_processing_info(self, sender):
+        if not sender:
+            return
+        launcher_config = self.config['launchers'][0]
+        dataset_config = self.config['datasets'][0]
+        sender.send_event('ac', 'platform', platform.system)
```
Comment (jkamelin) [inline tools/accuracy_checker/accuracy_checker/evaluators/model_evaluator.py:207]: Why not to use `send_telemetry_event` as in other cases?

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 3]
```diff
@@ -368,22 +377,25 @@ def build_arguments_parser():
...
     evaluator_kwargs = {}
-    if intermdeiate_metrics:
+    if args.intermediate_metrics_results:
         validate_print_interval(args.metrics_interval)
-        evaluator_kwargs['intermediate_metrics_results'] = intermdeiate_metrics
+        evaluator_kwargs['intermediate_metrics_results'] = args.intermediate_metrics_results
         evaluator_kwargs['metrics_interval'] = args.metrics_interval
         evaluator_kwargs['ignore_result_formatting'] = args.ignore_result_formatting
     evaluator_kwargs['store_only'] = args.store_only
+    send_telemetry_event("mode", "online" if not args.store_only else "offline")
```
Comment (jkamelin) [inline tools/accuracy_checker/accuracy_checker/main.py:392]:
```suggestion
    send_telemetry_event(tm, "mode", "online" if not args.store_only else "offline")
```

File: tools/accuracy_checker/accuracy_checker/evaluators/model_evaluator.py (hunk 2 above)
Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/evaluators/model_evaluator.py]: Thanks, fixed

**Review by vladimir-dudnik — APPROVED** [review]

**Review by Anna-Mironova — APPROVED** [review]

File: tools/accuracy_checker/accuracy_checker/annotation_converters/convert.py [hunk 4]
```diff
@@ -231,12 +242,14 @@ def main():
     out_dir = args.output_dir or Path.cwd()
 
     results = converter.convert()
+    send_telemetry_event(tm, 'annotation_conversion', 'finished')
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/convert.py:245]:
```suggestion
```

File: tools/accuracy_checker/accuracy_checker/dataset.py [hunk 5]
```diff
@@ -43,7 +43,7 @@
 from .utils import (
     JSONDecoderWithAutoConversion,
     read_json, read_yaml,
-    get_path, contains_all, set_image_metadata, OrderedSet, contains_any
+    get_path, contains_all, set_image_metadata, OrderedSet, contains_any,
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/dataset.py:46]:
```suggestion
    get_path, contains_all, set_image_metadata, OrderedSet, contains_any
```

File: tools/accuracy_checker/accuracy_checker/evaluators/base_evaluator.py [hunk 6]
```diff
@@ -50,3 +50,8 @@ def release(self):
     # reset progress for metrics calculation
     def reset(self):
         raise NotImplementedError
+
+    # helper for sending evaluation info
+    @staticmethod
+    def send_processing_info(sender):
+        pass
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/evaluators/base_evaluator.py:57]:
```suggestion
        return {}
```

**Pull request merged**